sdk/nodejs/docker.ts

```diff
@@ -260,18 +260,15 @@ async function buildAndPushImageWorkerAsync(
     }
 
     // If the container specified a cacheFrom parameter, first set up the cached stages.
-    let cacheFrom = Promise.resolve<string[] | undefined>(undefined);
+    let cacheFrom: string[] = [];
     if (pullFromCache) {
         const dockerBuild = <pulumi.UnwrappedObject<DockerBuild>>pathOrBuild;
         const cacheFromParam = (typeof dockerBuild.cacheFrom === "boolean" ? {} : dockerBuild.cacheFrom) || {};
-        cacheFrom = pullCacheAsync(baseImageName, cacheFromParam, repositoryUrl, logResource);
+        cacheFrom = await pullCacheAsync(baseImageName, cacheFromParam, repositoryUrl, logResource);
```

Comment:
second, concurrent execution of `docker pull` wasn't buying us anything.  We would immediately get to a point where we would just await this promise unilaterally before proceeding.  So it's not like this allowed us any extra performance.  It was just extra complexity.

Reply:
note: there is an opportunity for concurrency in that we can pull on all the different cache stages concurrently.  However, we already don't do that today.  And, if we did, it would be entirely in the body of pullCacheAsync.

I'm not going to make that change as i don't want that to cause any potential problems in case there's an issue with too many docker commands running concurrently, or interacting with some repository.